<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CSVAnalogSaver_Widget</class>
 <widget class="QWidget" name="CSVAnalogSaver_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>250</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label_description">
     <property name="text">
      <string>Description: Configure options for saving analog time series data to a CSV file. The CSV will contain two columns: Time and Data. Filename and parent directory are set by the calling widget.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_delimiter">
       <property name="text">
        <string>Delimiter:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="delimiter_combo">
       <item>
        <property name="text">
         <string>Comma</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Space</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Tab</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_line_ending">
       <property name="text">
        <string>Line Ending:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="line_ending_combo">
       <item>
        <property name="text">
         <string>LF (\n)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRLF (\r\n)</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_precision">
       <property name="text">
        <string>Precision (digits):</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_precision">
       <item>
        <widget class="QSpinBox" name="precision_spinbox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="value">
          <number>2</number> <!-- Default from CSVAnalogSaverOptions -->
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_precision_example">
         <property name="text">
          <string>e.g., 1.23</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_precision">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="QCheckBox" name="save_header_checkbox">
       <property name="text">
        <string>Save Header</string>
       </property>
       <property name="checked">
        <bool>true</bool> <!-- Default from CSVAnalogSaverOptions -->
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_header_text">
       <property name="text">
        <string>Header Text:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="header_text_edit">
       <property name="text">
        <string>Time,Data</string> <!-- Default from CSVAnalogSaverOptions -->
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPushButton" name="save_action_button">
     <property name="text">
      <string>Save Analog to CSV</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 